<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Intro: VisumHD Breast Cancer Dataset • highSpaClone</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Intro: VisumHD Breast Cancer Dataset">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">highSpaClone</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../data/index.html">Data</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Examples</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
    <li><a class="dropdown-item" href="../articles/Intro_to_highSpaClone.html">Intro: VisumHD Breast Cancer Dataset</a></li>
    <li><a class="dropdown-item" href="../articles/visiumhd_colon_code.html">VisiumHD Colorectal Cancer Datasets</a></li>
    <li><a class="dropdown-item" href="../articles/xenium_breast1_code.html">Xenium Breast Cancer Dataset 1</a></li>
    <li><a class="dropdown-item" href="../articles/xenium_breast2_code.html">Xenium Breast Cancer Dataset 2</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CZang409/highSpaClone/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Intro: VisumHD Breast Cancer Dataset</h1>
                        <h4 data-toc-skip class="author">Chenxuan Zang:
VisumHD Breast Cancer Dataset</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/CZang409/highSpaClone/blob/HEAD/vignettes/Intro_to_highSpaClone.Rmd" class="external-link"><code>vignettes/Intro_to_highSpaClone.Rmd</code></a></small>
      <div class="d-none name"><code>Intro_to_highSpaClone.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CZang409/highSpaClone" class="external-link">highSpaClone</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<div class="section level2">
<h2 id="loading-required-input-data">Loading required input data<a class="anchor" aria-label="anchor" href="#loading-required-input-data"></a>
</h2>
<p>highSpaClone requires three types of input data:</p>
<ul>
<li>Spatial transcriptomics count data (row: gene, column: cell).<br>
</li>
<li>Spatial location information: spatial coordinates for each cell,
with three columns (cell.id, x, y). The row names of location must match
with the column names of counts.).<br>
</li>
<li>Cell annotation information: cell type annotation information,
should contain two columns, (cell.id, cell.label). Can be a data.frame,
.csv or .txt file.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'./visiumhd_breast_data.rds'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="va">location</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">spatial_location</span></span>
<span><span class="va">annotation</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">annotation</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-highspaclone-object">Create highSpaClone object<a class="anchor" aria-label="anchor" href="#create-highspaclone-object"></a>
</h2>
<p>This step is to create a highSpaClone object. The most important
input arguments are:</p>
<ul>
<li>
<code>counts</code>: Gene-by-cell raw count matrix. Ensure to
contain the row/column names.</li>
<li>
<code>location</code>: A data.frame or matrix of spatial coordinates
for each cell with three columns representing the cell id, x and y
coordinates of the spatial location (The column names must contain
“cell.id”, “x” and “y”).</li>
<li>
<code>annotations_file</code>: Cell type annotation file with two
columns representing the cell id and cell type/cluster of each cell.
Must be a data.frame, .csv or .txt file.</li>
<li>
<code>gene_order_file</code>: The gene order file, contains
chromosome, start, and stop position for each gene. If not provided, the
function uses the built-in hg38 annotation. Users may upload/provide a
custom file to replace the default.</li>
<li>
<code>min_avg_expression</code>: Minimum average expression per gene
(default: 0.01).</li>
<li>
<code>min_gene_counts</code>: Minimum total counts per cell
(default: 100).</li>
<li>
<code>project</code>: Project name.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createObject.html">createObject</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">counts</span>,</span>
<span>                    location<span class="op">=</span><span class="va">location</span>,</span>
<span>                    annotations_file <span class="op">=</span> <span class="va">annotation</span>,</span>
<span>                    gene_order_file<span class="op">=</span><span class="cn">NULL</span>,</span>
<span>                    min_avg_expression<span class="op">=</span><span class="fl">0.01</span>,</span>
<span>                    min_gene_counts<span class="op">=</span><span class="fl">100</span>,</span>
<span>                    project<span class="op">=</span><span class="st">'visumhd.breast'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Filtered genes based on average expression threshold: 6107 genes removed.</span></span>
<span><span class="co">#&gt; Remaining genes after filtering: 11978 genes.</span></span>
<span><span class="co">#&gt; Filtered cells based on minimum gene counts threshold: 385 cells removed.</span></span>
<span><span class="co">#&gt; Remaining cells after filtering: 999 cells.</span></span>
<span><span class="co">#&gt; Removed 0 genes that did not match the gene order file.</span></span>
<span><span class="co">#&gt; Remaining 11609 genes after matching with the gene order file.</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="main-function">Main function<a class="anchor" aria-label="anchor" href="#main-function"></a>
</h2>
<div class="section level3">
<h3 id="smooth-by-chromosome">Smooth by chromosome<a class="anchor" aria-label="anchor" href="#smooth-by-chromosome"></a>
</h3>
<p>This step is to smooth gene expression by gene order within each
chromosome, which improves CNV signal continuity and suppresses
gene-specific noise. The most important input arguments are:</p>
<ul>
<li>
<code>obj</code>: highSpaClone object.</li>
<li>
<code>window_size</code>: Size of the running-mean window (in genes)
used within each chromosome (default: 101). Must be an integer.</li>
<li>
<code>step</code>: Number of genes in a bin (default: 50). Must be
an integer.</li>
<li>
<code>use_chunk</code>: Whether to split cells into smaller subsets
(chunks) during computation (default: TRUE).</li>
<li>
<code>chunk_size</code>: The number of cells per chunk when chunking
is enabled (default: 5000).</li>
<li>
<code>parallel</code>: If <code>TRUE</code>, chunks are processed in
parallel to accelerate runtime (default: FALSE).</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># smooth gene expression by chromosome</span></span>
<span><span class="va">smooth_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/smooth_expr.html">smooth_expr</a></span><span class="op">(</span>obj <span class="op">=</span> <span class="va">obj</span>, </span>
<span>                          window_size <span class="op">=</span> <span class="fl">101</span>,</span>
<span>                          step <span class="op">=</span> <span class="fl">50</span>,</span>
<span>                          use_chunk <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                          chunk_size <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                          parallel <span class="op">=</span> <span class="cn">T</span> </span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Normalization time:  1.348653  seconds</span></span>
<span><span class="co">#&gt; Working with sparse matrix format (keeping sparse)</span></span>
<span><span class="co">#&gt; Total cells:  118697 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === Chunking mode enabled ===</span></span>
<span><span class="co">#&gt; Processing in chunks of size  5000 ...</span></span>
<span><span class="co">#&gt; Total chunks:  24 </span></span>
<span><span class="co">#&gt; === Parallel processing enabled ===</span></span>
<span><span class="co">#&gt; Requested cores:  4  (max_cores limit:  4 )</span></span>
<span><span class="co">#&gt; Estimated memory usage: ~ 13.26  GB</span></span>
<span><span class="co">#&gt; Windows system detected: using parLapply</span></span>
<span><span class="co">#&gt; Using  4  cores</span></span>
<span><span class="co">#&gt; Processing chunk 5 of 24</span></span>
<span><span class="co">#&gt; Processing chunk 10 of 24</span></span>
<span><span class="co">#&gt; Processing chunk 15 of 24</span></span>
<span><span class="co">#&gt; Processing chunk 20 of 24</span></span>
<span><span class="co">#&gt; Parallel processing completed.</span></span>
<span><span class="co">#&gt; Combining results...</span></span>
<span><span class="co">#&gt; Combining dense matrices...</span></span>
<span><span class="co">#&gt; Result class:  matrix </span></span>
<span><span class="co">#&gt; Result dimensions:  118697  x  190 </span></span>
<span><span class="co">#&gt; Calculating chromosome positions...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Storing results in object...</span></span>
<span><span class="co">#&gt; Results successfully stored.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; Total running time:   3 minutes</span></span>
<span><span class="co">#&gt; ========================================</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="identify-tumor-cells-optional">Identify tumor cells (optional)<a class="anchor" aria-label="anchor" href="#identify-tumor-cells-optional"></a>
</h3>
<p>The <code><a href="../reference/FindTumor.html">FindTumor()</a></code> function identifies tumor cells based on
CNV profiles when no prior tumor annotations are available. This
function performs reference-guided tumor segmentation using the smoothed
highSpaClone object.</p>
<p>It requires users to specify the normal (reference) cell population,
which can be provided in either of the following forms:</p>
<ul>
<li>
<code>ref</code>: A character vector of reference cell type
labels.</li>
<li>
<code>ref.id</code>: A character vector of reference cell IDs.</li>
</ul>
<p>Only one of these arguments needs to be provided.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Tumor segmentation</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindTumor.html">FindTumor</a></span><span class="op">(</span><span class="va">smooth_obj</span>,</span>
<span>                 ref<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B Cells"</span>, <span class="st">"CD4+ T Cells"</span>, <span class="st">"CD8+ T Cells"</span>, </span>
<span>                 <span class="st">"Macrophages 1"</span>, <span class="st">"Macrophages 2"</span>, <span class="st">"Endothelial"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; Start Tumor Annotation...</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; CNV matrix dimensions:  190  x  118697 </span></span>
<span><span class="co">#&gt; Reference cell types:  B Cells, CD4+ T Cells, CD8+ T Cells, Macrophages, Endothelial </span></span>
<span><span class="co">#&gt; Number of reference cells:  23995 </span></span>
<span><span class="co">#&gt; Iteration  1  /  500 </span></span>
<span><span class="co">#&gt; Iteration  25  /  500 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Converged at iteration  30 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Iteration time:  6.5  minutes</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; Tumor annotation completed!</span></span>
<span><span class="co">#&gt; Total time:  6.5  minutes</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span></span>
<span><span class="co"># Visualization</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#ebe5c2"</span>, <span class="st">"#D57358"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/spatialplot.html">spatialplot</a></span><span class="op">(</span><span class="va">obj</span>, colors<span class="op">=</span><span class="va">colors</span>, point_size <span class="op">=</span> <span class="fl">0.01</span>, use_coord_flip<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="Intro_to_highSpaClone_files/figure-html/unnamed-chunk-5-1.png" width="2187.5"></p>
</div>
<div class="section level3">
<h3 id="suggest-the-number-of-subclones-optional">Suggest the number of subclones (optional)<a class="anchor" aria-label="anchor" href="#suggest-the-number-of-subclones-optional"></a>
</h3>
<p>The number of tumor subclones must be pre-specified. If there is no
prior knowledge, we provide the function <code>suggest_k</code>, which
recommends an appropriate value for K based on a Silhouette score that
integrates both CNV and spatial information. The most important input
arguments are:</p>
<ul>
<li>
<code>obj</code>: smoothed highSpaClone object.</li>
<li>
<code>ref</code>: A character vector of reference cell type
labels.</li>
<li>
<code>tumor</code>: A character vector of tumor cell type
labels.</li>
<li>
<code>k_range</code>: Integer vector of K values to evaluate.</li>
<li>
<code>n_sub</code>: Max number of cells to subsample for evaluation
(default: 5000).</li>
<li>
<code>alpha</code>: A number bewtween 0 and 1. Control the weight of
CNV in clustering.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Suggest the number of subclones</span></span>
<span><span class="fu"><a href="../reference/suggest_k.html">suggest_k</a></span><span class="op">(</span></span>
<span>    obj<span class="op">=</span><span class="va">smooth_obj</span>,</span>
<span>    ref<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B Cells"</span>, <span class="st">"CD4+ T Cells"</span>, <span class="st">"CD8+ T Cells"</span>, </span>
<span>          <span class="st">"Macrophages 1"</span>, <span class="st">"Macrophages 2"</span>, <span class="st">"Endothelial"</span><span class="op">)</span>,</span>
<span>    tumor<span class="op">=</span><span class="st">"Invasive Tumor"</span>,</span>
<span>    k_range <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">8</span>,</span>
<span>    n_sub <span class="op">=</span> <span class="fl">500</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>    seed <span class="op">=</span> <span class="fl">123</span>,</span>
<span>    out_dir <span class="op">=</span> <span class="st">"figs_k"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Suggested K by average silhouette = 3 (mean=0.244)</span></span></code></pre></div>
<p><img src="Intro_to_highSpaClone_files/figure-html/unnamed-chunk-6-1.png" width="2187.5"></p>
</div>
<div class="section level3">
<h3 id="spatial-cnv-inference-and-subclone-detection">Spatial CNV inference and subclone detection<a class="anchor" aria-label="anchor" href="#spatial-cnv-inference-and-subclone-detection"></a>
</h3>
<p>The <code><a href="../reference/FindClone.html">FindClone()</a></code> function performs spatial CNV inference
and tumor subclone identification based on CNV patterns, enhanced by
spatial regularization. After tumor cells are identified, this function
clusters them into subclonal populations that share similar CNV states
while also preserving spatial continuity.</p>
<p>Users must define a reference (normal) cell population using
either:</p>
<ul>
<li>
<code>ref</code>: A character vector of reference (normal) cell type
labels, or</li>
<li>
<code>ref.id</code>: A character vector of reference cell IDs.</li>
</ul>
<p>All non-reference cells are treated as tumor cells by default.
Alternatively, users may explicitly specify the tumor population
via:</p>
<ul>
<li>
<code>tumor</code>: A character vector of tumor cell type labels,
or</li>
<li>
<code>tumor.id</code>: A character vector of tumor cell IDs.</li>
</ul>
<p>The parameter <code>lambda</code> controls the strength of spatial
smoothing, encouraging neighboring cells to remain in the same subclone
if supported by CNV evidence. The number of subclones to infer is set by
<code>K</code>, allowing flexible modeling of tumor heterogeneity.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Spatial CNV inference and tumor subclone detection</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindClone.html">FindClone</a></span><span class="op">(</span>obj<span class="op">=</span><span class="va">smooth_obj</span>,</span>
<span>                 ref<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B Cells"</span>, <span class="st">"CD4+ T Cells"</span>, <span class="st">"CD8+ T Cells"</span>, </span>
<span>                      <span class="st">"Macrophages 1"</span>, <span class="st">"Macrophages 2"</span>, <span class="st">"Endothelial"</span><span class="op">)</span>,</span>
<span>                 tumor<span class="op">=</span><span class="st">"Invasive Tumor"</span>,</span>
<span>                 K<span class="op">=</span><span class="fl">3</span>,</span>
<span>                 lambda<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; Running highSpaClone...</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [Step 1] Preparing data...</span></span>
<span><span class="co">#&gt; CNV matrix dimensions (genes x cells):  190  x  118697 </span></span>
<span><span class="co">#&gt; Reference cell types (ref):  B Cells, CD4+ T Cells, CD8+ T Cells, Macrophages, Endothelial </span></span>
<span><span class="co">#&gt; Reference cell number:  23995 </span></span>
<span><span class="co">#&gt; Tumor cell number:  65579 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [Step 2] Initializing matrices...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [Step 3] Starting iterative optimization...</span></span>
<span><span class="co">#&gt; Iteration  1  /  500 </span></span>
<span><span class="co">#&gt; Iteration  25  /  500 </span></span>
<span><span class="co">#&gt; Iteration  50  /  500 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Converged at iteration  57 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Iteration time:  8.07  minutes</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [Step 4] Storing results...</span></span>
<span><span class="co">#&gt; Results stored in object.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span><span class="co">#&gt; CNV inference and tumor subclone detection completed!</span></span>
<span><span class="co">#&gt; Total time:  8.07  minutes</span></span>
<span><span class="co">#&gt; ========================================</span></span>
<span></span>
<span><span class="co"># Visualization</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#E64B35"</span>, <span class="st">"#4DBBD5"</span>, <span class="st">"#8491B4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/spatialplot.html">spatialplot</a></span><span class="op">(</span><span class="va">obj</span>, colors<span class="op">=</span><span class="va">colors</span>, point_size <span class="op">=</span> <span class="fl">0.01</span>, use_coord_flip<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="Intro_to_highSpaClone_files/figure-html/unnamed-chunk-7-1.png" width="2187.5"></p>
</div>
<div class="section level3">
<h3 id="visualize-cnv-heatmap">Visualize CNV heatmap<a class="anchor" aria-label="anchor" href="#visualize-cnv-heatmap"></a>
</h3>
<p>Visualize subclonal CNV patterns by heatmap:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize CNV heatmap</span></span>
<span><span class="va">log.cnv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">cnv.data</span><span class="op">)</span></span>
<span><span class="va">log.cnv</span><span class="op">[</span><span class="va">log.cnv</span> <span class="op">&gt;</span> <span class="op">-</span><span class="fl">0.5</span> <span class="op">&amp;</span> <span class="va">log.cnv</span> <span class="op">&lt;</span> <span class="fl">0.5</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span></span>
<span><span class="va">breasthd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  cnv.data   <span class="op">=</span> <span class="va">log.cnv</span>,</span>
<span>  annotation <span class="op">=</span> <span class="va">obj</span><span class="op">@</span><span class="va">cluster</span>,</span>
<span>  chr_pos    <span class="op">=</span> <span class="va">obj</span><span class="op">@</span><span class="va">chr_pos</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/cnv_heatmap.html">cnv_heatmap</a></span><span class="op">(</span></span>
<span>  cnv.obj<span class="op">=</span><span class="va">breasthd</span>,</span>
<span>  groupby <span class="op">=</span> <span class="st">"cell.label"</span>,</span>
<span>  min_value  <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>,  </span>
<span>  max_value  <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  show    <span class="op">=</span> <span class="cn">TRUE</span>,       </span>
<span>  save    <span class="op">=</span> <span class="cn">TRUE</span>,         </span>
<span>  outfile <span class="op">=</span> <span class="st">"breasthd_heatmap.png"</span> </span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Intro_to_highSpaClone_files/figure-html/unnamed-chunk-8-1.png" width="2187.5"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Chenxuan Zang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
